scanner/matcher.go

```diff
@@ -127,3 +131,94 @@ func (m MatchIssuerRegex) CertificateMatches(c *x509.Certificate) bool {
 func (m MatchIssuerRegex) PrecertificateMatches(p *ct.Precertificate) bool {
 	return m.PrecertificateIssuerRegex.FindStringIndex(p.TBSCertificate.Issuer.CommonName) != nil
 }
+
+// RawMatcher describes how to match log entries, based on the raw log entry information;
+// clients should implement this interface to perform their own match criteria.
+type RawMatcher interface {
+	LeafMatches(*ct.LeafEntry) bool
+}
+
+// MatchParseErrs is a RawMatcher which will match any Certificate or Precertificate that
+// triggered an error on parsing.
+type MatchParseErrs struct {
```

Comment:
This name also seems a bit odd but I don't have a better suggestion so will pass on it.

Reply:
Changed `MatchParseErrs` -> `CertParseFailMatcher`, `MatchValidationErrs` -> `CertVerifyFailMatcher`